modules/rasBidAdapter.js

```diff
@@ -0,0 +1,163 @@
+    height: ad.height || 0,
+    ttl: 300,
+    creativeId: ad.adid ? parseInt(ad.adid.split(',')[2], 10) : 0,
+    netRevenue: true,
+    currency: ad.currency || 'USD',
+    dealId: null,
+    meta: {
+      mediaType: BANNER
+    },
+    ad: ad.html || null
+  };
+};
+
+const getContextParams = (bidRequests) => {
+  const bid = bidRequests[0];
+  const { params } = bid;
+  const requestParams = {
+    nid: encodeURIComponent(params.network),
+    site: params.site,
+    area: params.area,
+    cre_format: 'html',
+    systems: 'das',
+    kvprver: VERSION,
+    ems_url: 1,
+    bid_rate: 1,
+    ...parseParams(params)
+  };
+  return Object.keys(requestParams).map((key) => encodeURIComponent(key) + '=' + encodeURIComponent(requestParams[key])).join('&');
+};
+
+const getSlots = (bidRequests) => {
+  let queryString = '';
+  const batchSize = bidRequests.length;
+  for (let i = 0; i < batchSize; i++) {
+    const adunit = bidRequests[i];
+    const { slot } = adunit.params;
+    const sizes = parseSizesInput(getAdUnitSizes(adunit)).join(',');
+    queryString += `&slot${i}=${encodeURIComponent(slot)}&id${i}=${encodeURIComponent(adunit.bidId)}&composition${i}=CHILD`;
+    queryString += sizes.length ? `&iusizes${i}=${encodeURIComponent(sizes)}` : ''
+  }
+  return queryString;
+};
+
+const getGdprParams = (bidderRequest) => {
+  const gdprApplies = deepAccess(bidderRequest, 'gdprConsent.gdprApplies');
+  let consentString = deepAccess(bidderRequest, 'gdprConsent.consentString');
+  let queryString = '';
+  if (gdprApplies !== undefined) {
+    queryString += `&gdpr_applies=${encodeURIComponent(gdprApplies)}`;
+  }
+  if (consentString !== undefined) {
+    queryString += `&euconsent=${encodeURIComponent(consentString)}`;
+  }
+  return queryString;
+};
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [BANNER],
+
+  isBidRequestValid: function (bidRequest) {
+    if (!bidRequest || !bidRequest.params || typeof bidRequest.params !== 'object') {
+      return;
+    }
+    const { params } = bidRequest;
+    return Boolean(params.network && params.site && params.area);
+  },
+
+  buildRequests: function (bidRequests, bidderRequest) {
+    const slotsQuery = getSlots(bidRequests);
+    const contextQuery = getContextParams(bidRequests);
+    const gdprQuery = getGdprParams(bidderRequest);
+    const bidIds = bidRequests.map((bid) => ({ slot: bid.params.slot, bidId: bid.bidId }));
+    const network = bidRequests[0].params.network;
+    return [{
+      method: 'GET',
+      url: GET_ENDPOINT(network) + contextQuery + slotsQuery + gdprQuery,
+      bidIds: bidIds
+    }];
+  },
+
+  interpretResponse: function (serverResponse, bidRequest) {
+    const response = serverResponse.body;
+    if (!response || !response.ads || response.ads.length === 0) {
+      return [];
+    }
+    return response.ads.map(buildBid).filter((bid) => !isEmpty(bid));
+  },
+
+  getUserSyncs: function (syncOptions, serverResponses) {
```

Comment:
Functions getUserSyncs, onTimeout, onBidWon, onSetTargeting can be removed If you are not planning on implementing them with this PR.

Reply:
The functions have been removed